custom_scripts/py_scripts/db_scanner/db_scanner.py

```diff
@@ -0,0 +1,255 @@
+                    if document[d] not in ["", None, [], {}]:
+                        document_list.update({f"{d}": document[d]})
+                    break
+        else:
+            if isinstance(document[d], list):
+                for item in document[d]:
+                    # print(f"item is: {item}, type is: {type(item)}")
+                    if type(item) == list:
+                        for value in item:
+                            if isinstance(value, dict):
+                                result_list = find_values_by_key(value)
+                                # print(f"result list: {result_list}")
+                                if result_list is not None:
+                                    document_list.update({value: result_list})
+                    if isinstance(item, dict):
+                        result_list = find_values_by_key(item)
+                        # print(f"result list: {result_list}")
+                        if result_list is not None:
+                            document_list.update({item: result_list})
+            if isinstance(document[d], dict):
+                result_list = find_values_by_key(document[d])
+                # print(f"doc type is {type(document[d])} and result's is {type(result_list)}")
+                # print(f"doc is {document[d]} and result is {result_list}")
+                if result_list is not None:
+                    document_list.update(result_list)
+    if len(document_list) == 0:
+        return None
+    # new_list = list(k for k, _ in groupby(document_list))
+    # print("ok, here we are: ", document_list)
+    return document_list
+
+
+# TODO: add regular expressions for finding data by value
+# TODO: restrict the connection's timeout in some ways
+def connect_to_db(ip_value: str) -> (dict or None, Exception or None):
+    """
+    Connect to PyMongoDB and look for possible critical information
+    :param ip_value: IP-address of host without authentication
+    :return: founded info and caught exception, if something occurred
+    """
+    port = 27017
+    try:
+        client = pymongo.MongoClient(
+            ip_value, port, serverSelectionTimeoutMS=2500, socketTimeoutMS=300000
+        )  # , heartbeatFrequencyMS=2500
+    except ServerSelectionTimeoutError as err:
+        print(f"Host was unreachable: {ip_value} with {err}")
+        return None, err
+    if client is not None:
+        try:
+            db_names = client.list_database_names()
+        except (
+            ServerSelectionTimeoutError,
+            ConfigurationError,
+            NetworkTimeout,
+            OperationFailure,
+            KeyError,
+        ) as err:
+            print(f"Host was unreachable: {ip_value} with {err}")
+            return None, err
+        if db_names is None:
+            return None
+        host_dict = {
+            "host": ip_value,
+            "port": port,
+            "version": client.server_info()["version"],
```

Comment:
`"version"` is always existed?
If not, this part will fail.

Better to check it somehow, for example, try:
```python3
server_version = client.server_info()
version = server_version.get("version", "Version unknown")
```